hermes/bin/tests/wasm-integration/main.rs

```diff
@@ -0,0 +1,128 @@
+//! Integration tests for Hermes WASM components
+
+// SEE: https://docs.rs/libtest-mimic/latest/libtest_mimic/index.html
+
+/// A parameter identifier specifying the directory for placing test WebAssembly
+/// components.
+const ENV_MODULE_DIR: &str = "TEST_WASM_MODULE_DIR";
+/// A standard value assigned to `ENV_MODULE_DIR` when it's not specified.
+const DEFAULT_TEST_COMPONENT_DIR: &str = "tests/test-components";
+
+use std::{env, error::Error, ffi::OsStr, fs, path::Path};
+
+use hermes::{
+    runtime_extensions::hermes::integration_test::event::execute_event, wasm::module::Module,
+};
+use libtest_mimic::{Arguments, Failed, Trial};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args = Arguments::from_args();
+    let tests = collect_tests()?;
+    libtest_mimic::run(&args, tests).exit();
+}
+
+/// Creates as many tests as required for each `.wasm` file in the current directory or
+/// sub-directories of the current directory.
+fn collect_tests() -> Result<Vec<Trial>, Box<dyn Error>> {
+    #[allow(clippy::missing_docs_in_private_items)]
+    fn visit_dir(path: &Path, tests: &mut Vec<Trial>) -> Result<(), Box<dyn Error>> {
+        let entries: Vec<_> = fs::read_dir(path)?
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .map(|entry| entry.file_type().map(|file_type| (file_type, entry.path())))
+            .collect::<Result<_, _>>()?;
+        let wasm_file_paths: Vec<_> = entries
+            .iter()
+            .filter_map(|(file_type, path)| {
+                (file_type.is_file() && path.extension() == Some(OsStr::new("wasm")))
+                    .then_some(path)
+            })
+            .collect();
+        let dir_paths: Vec<_> = entries
+            .iter()
+            .filter_map(|(file_type, path)| file_type.is_dir().then_some(path))
+            .collect();
+
+        // process `.wasm` files
+        for file_path in wasm_file_paths {
+            let name = file_path.strip_prefix(path)?.display().to_string();
+
+            // Execute the wasm tests to get their name
+            // Load WASM module in the executor.
+            let wasm_buf = fs::read(file_path)?;
+            let mut module = Module::new(name.clone(), &wasm_buf)?;
+
+            // Run the tests in a loop until no more tests.
+            for i in 0..32 {
```

Comment:
This is a limit to prevent a bad wasm module from locking up the test runner.
We rely on the test runner inside the wasm to tell us there are no more tests if it fails to do so, we will try forever.
We have said currently that we will limit it to a maximum of 32 tests within a single wasm module, we can change this option later, or make it configurable but that's out of scope for this PR.